sui/tests/shared_objects_tests.rs

```diff
@@ -1,55 +1,58 @@
-use sui_types::error::{SuiError, SuiResult};
-use sui_types::messages::CallArg;
-use sui_types::messages::Transaction;
-use sui_types::messages::TransactionInfoResponse;
-use sui_types::messages::{ConsensusTransaction, ExecutionStatus};
-use sui_types::object::Object;
-use sui_types::serialize::{
-    deserialize_message, deserialize_transaction_info, serialize_cert,
-    serialize_consensus_transaction, SerializedMessage,
+use sui_core::authority_client::{AuthorityAPI, NetworkAuthorityClient};
+use sui_network::network::NetworkClient;
+use sui_types::{
+    base_types::ObjectRef,
+    error::SuiResult,
+    messages::{
+        CallArg, ConfirmationTransaction, ConsensusTransaction, ExecutionStatus, Transaction,
+        TransactionInfoResponse,
+    },
+    object::Object,
+};
+use test_utils::{
+    authority::{spawn_test_authorities, test_authority_configs},
+    messages::{
+        make_certificates, move_transaction, parse_package_ref, publish_move_package_transaction,
+        test_shared_object_transactions,
+    },
+    objects::{test_gas_objects, test_shared_object},
 };
-use test_utils::authority::{spawn_test_authorities, test_authority_configs};
-use test_utils::messages::{make_certificates, move_transaction, publish_move_package_transaction};
-use test_utils::messages::{parse_package_ref, test_shared_object_transactions};
-use test_utils::objects::{test_gas_objects, test_shared_object};
-use tokio::net::TcpStream;
-use tokio_util::codec::Framed;
-use tokio_util::codec::LengthDelimitedCodec;
-
-/// Send bytes to a Sui authority.
-async fn transmit(transaction: Bytes, config: &AuthorityPrivateInfo) -> SerializedMessage {
-    let authority_address = format!("{}:{}", config.host, config.port);
-    let stream = TcpStream::connect(authority_address).await.unwrap();
-    let mut connection = Framed::new(stream, LengthDelimitedCodec::new());
-
-    connection.send(transaction).await.unwrap();
-    let bytes = connection.next().await.unwrap().unwrap();
-    deserialize_message(&bytes[..]).unwrap()
-}
 
 /// Submit a certificate containing only owned-objects to all authorities.
 async fn submit_single_owner_transaction(
     transaction: Transaction,
     configs: &[AuthorityPrivateInfo],
 ) -> Vec<TransactionInfoResponse> {
     let certificate = make_certificates(vec![transaction]).pop().unwrap();
-    let serialized = Bytes::from(serialize_cert(&certificate));
+    let txn = ConfirmationTransaction { certificate };
 
     let mut responses = Vec::new();
     for config in configs {
-        let bytes = transmit(serialized.clone(), config).await;
-        let reply = deserialize_transaction_info(bytes).unwrap();
+        let client = get_client(config);
+        let reply = client
+            .handle_confirmation_transaction(txn.clone())
+            .await
+            .unwrap();
         responses.push(reply);
     }
     responses
 }
 
+fn get_client(config: &AuthorityPrivateInfo) -> NetworkAuthorityClient {
```

Comment:
Lets use the AuthorityInfo here that does not contain private keys.

Reply:
This is a test so its not as important. But I do have plans on dramatically improving our test infra from a networking perspective that will have to wait for a future PR